### PR TITLE
feat(v3): update @angular v6.1.10

### PR DESCRIPTION
**Pull request by rdlabo**
#### Short description of what this resolves:
Update dependencies package. We still need Ionic v3. and if update ionic v3 to v4, I think we should update angular package first.
So I want to update Ionic v3 to correspond Angular 6(This is LTS!）.

https://github.com/ionic-team/ionic/pull/15455

#### Changes proposed in this pull request:
update @angular package.
update rxjs6

**Fixes**: #

### CODE REVIEW

**Comment by jackgllghr** [comment]
Please merge this when stable, we're keen to upgrade to Ionic 4, so anything that makes the migration path one step less is a big plus in our books. 

I think an ideal path would be:
3.10 - Angular 6 support with RxJS 6 and RxJS-Compat
3.11 - SCSS to support using styleUrls in components with migration script
3.12 - Support for Angular Router 

Then all that would be left is the API updates/regular code cleanup/scaffolding during the migration, which can't be avoided. Am I missing anything?

File: package.json [hunk 1]
```diff
@@ -26,20 +26,20 @@
...
+    "@angular/common": "6.1.10",
+    "@angular/compiler": "6.1.10",
+    "@angular/compiler-cli": "6.1.10",
+    "@angular/core": "6.1.10",
+    "@angular/forms": "6.1.10",
+    "@angular/http": "6.1.10",
+    "@angular/platform-browser": "6.1.10",
+    "@angular/platform-browser-dynamic": "6.1.10",
+    "ionicons": "3.0.0",
+    "rxjs": "6.2.2",
```
Comment (jackgllghr) [inline package.json]: May need rxjs-compat here

File: tsconfig.json [hunk 2]
```diff
@@ -8,7 +8,7 @@
     "module": "commonjs",
     "moduleResolution": "node",
     "noImplicitAny": true,
-    "noUnusedLocals": true,
+    "noUnusedLocals": false,
```
Comment (jackgllghr) [inline tsconfig.json]: Why was this turned off?

Comment (rdlabo) [reply tsconfig.json]: This pull request include typescript@2.9. Option of noUnusedLocals is renovated.

File: package.json (hunk 1 above)
Comment (rdlabo) [reply package.json]: This pull request don't require rxjs-compat.

Comment (longgt) [reply package.json]: When I tried to upgrade to Angular 5, Ionic source code was compiled with duplicated comments in all of functions.
I don't have any experienced with Angular 6 but the above issue forced me to stop upgrading Angular dependencies.

**Comment by longgt** [comment]
@rdlabo In nav-controller-base.ts, Ionic is using `ReflectiveInjector` which was deprecated since Angular 6.
So I think you should add the changes from `ReflectiveInjector` to `StaticInjector` to this pull request.
https://blog.angularindepth.com/angular-introduces-staticinjector-should-you-care-4e059eca030c

**Comment by adamduren** [comment]
Currently maintaining a v3 app and curious if this is ever going to happen?

It's becoming increasingly hard to figure out the right combination of dependencies when installing a new package and many packages have bug fixes in later versions that do not support `@angular@5.0.0`. I know upgrading to v4 is recommended, however that is not feasible for every project depending on their constraints.

**Comment by rdlabo** [comment]
Sorry. I don't maintenance. I will close this pull request. Thanks.

**Pull request closed without merging**